From 662670dc0dcd84526c29a99af4f9a8cdb053ab42 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Mon, 24 Jul 2017 18:01:40 -0700
Subject: [PATCH] netgear: restart driver on failure

Also wait until the network is up, in case we're monitoring a SNMP UPS.
---
 scripts/systemd/nut-driver.service.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scripts/systemd/nut-driver.service.in b/scripts/systemd/nut-driver.service.in
index 3f97d13f..7a22be15 100644
--- a/scripts/systemd/nut-driver.service.in
+++ b/scripts/systemd/nut-driver.service.in
@@ -1,11 +1,14 @@
 [Unit]
 Description=Network UPS Tools - power device driver controller
 After=local-fs.target network.target systemd-udev-settle.service
-Wants=systemd-udev-settle.service
+Wants=systemd-udev-settle.service network-online.target
+After=network-online.target
 StopWhenUnneeded=yes
 
 [Service]
 ExecStart=@SBINDIR@/upsdrvctl start
 ExecStop=@SBINDIR@/upsdrvctl stop
 Type=forking
+Restart=on-failure
+RestartSec=2
 
-- 
2.13.0

